docker/src/main/java/org/jclouds/docker/compute/strategy/DockerComputeServiceAdapter.java

```diff
@@ -237,15 +238,22 @@ public DockerComputeServiceAdapter(DockerApi api) {
    }
 
    @Override
-   public Image getImage(final String imageId) {
-      // less efficient than just inspectImage but listImages return repoTags
-      return find(listImages(), new Predicate<Image>() {
+   public Image getImage(final String imageIdOrName) {
+      if (imageIdOrName.startsWith("sha256")) {
+         // less efficient than just inspectImage but listImages return repoTags
+         Image image = find(listImages(), new Predicate<Image>() {
+            @Override
+            public boolean apply(Image input) {
+               // Only attempt match on id as we should try to pull again anyway if using name
+               return input.id().equals(imageIdOrName);
+            }
+         }, null);
+         return image;
+      }
 
-         @Override
-         public boolean apply(Image input) {
-            return input.id().equals(imageId);
-         }
-      }, null);
+      // Image is not cached or getting image by name so try to pull it
+      api.getImageApi().createImage(CreateImageOptions.Builder.fromImage(imageIdOrName));
+      return api.getImageApi().inspectImage(imageIdOrName);
```

Comment:
as per the comment at line 243, api.getImageApi().listImages and api.getImageApi().inspectImage are slightly different: listImages returns also `repoTags` which are useful at [here](https://github.com/jclouds/jclouds-labs/blob/e3c6a6f5a7b26badaa385a2ba626fcb9e386761a/docker/src/main/java/org/jclouds/docker/compute/functions/ImageToImage.java#L49-L49)

can you change the impl to populate `repoTags` even when the image gets pulled?